packages/replay-next/src/suspense/ExceptionsCache.ts

```diff
@@ -1,74 +1,43 @@
+import { createCache } from "suspense";
 
-import {
-  AnalysisInput,
-  AnalysisResultWrapper,
-  SendCommand,
-  getFunctionBody,
-} from "protocol/evaluation-utils";
+import { ReplayClientInterface } from "shared/client/types";
 
 import { createInfallibleSuspenseCache } from "../utils/suspense";
-import { AnalysisParams, RemoteAnalysisResult, createAnalysisCache } from "./AnalysisCache";
+import { createAnalysisCache } from "./AnalysisCache";
+import { cachePauseData } from "./PauseCache";
 
 export type UncaughtException = PointDescription & {
   type: "UncaughtException";
 };
 
-// Variables in scope in an analysis
-declare let sendCommand: SendCommand;
-declare let input: AnalysisInput;
-
-// This function will be evaluated on the server!
-function exceptionsMapper(): AnalysisResultWrapper<RemoteAnalysisResult>[] {
-  const finalData: Required<PauseData> = { frames: [], scopes: [], objects: [] };
-
-  function addPauseData({ frames, scopes, objects }: PauseData) {
-    finalData.frames.push(...(frames || []));
-    finalData.scopes.push(...(scopes || []));
-    finalData.objects.push(...(objects || []));
-  }
-
-  const { pauseId, point, time } = input;
-
-  const { data: exceptionValueData, exception } = sendCommand("Pause.getExceptionValue", {});
-  addPauseData(exceptionValueData);
-
-  const { data: allFramesData, frame: frameId } = sendCommand("Pause.getTopFrame", {});
-  addPauseData(allFramesData);
-
-  const topFrame = finalData.frames.find(f => f.frameId === frameId)!;
-  const location = topFrame.location;
-
-  return [
-    {
-      key: time,
-      value: {
-        data: finalData,
-        location,
-        pauseId,
-        point,
-        time,
-        values: [exception!],
-      },
-    },
-  ];
-}
-
 export const exceptionsCache = createAnalysisCache<UncaughtException, []>(
   "ExceptionsCache",
-  () => analysisParams,
+  () => "",
+  (client, begin, end) => client.findPoints(pointSelector, { begin, end }),
+  () => ({ selector: pointSelector, expression: "[]" }),
   transformPoint
 );
 
-const analysisParams: AnalysisParams = {
-  exceptions: true,
-  mapper: getFunctionBody(exceptionsMapper),
-};
+export const getInfallibleExceptionPointsSuspense = createInfallibleSuspenseCache(
+  exceptionsCache.pointsIntervalCache.read
+);
+
+const pointSelector: PointSelector = { kind: "exceptions" };
 
 function transformPoint(pointDescription: PointDescription): UncaughtException {
   return { type: "UncaughtException", ...pointDescription };
 }
 
-export const getInfallibleExceptionPointsSuspense = createInfallibleSuspenseCache(
-  exceptionsCache.pointsIntervalCache.read
-);
+export const exceptionValueCache = createCache<
+  [client: ReplayClientInterface, pauseId: PauseId],
+  Value | undefined
+>({
+  debugLabel: "ExceptionValueCache",
+  getKey: ([client, pauseId]) => pauseId,
+  load: async ([client, pauseId]) => {
+    const result = await client.getExceptionValue(pauseId);
```

Comment:
Could you check how this behaves on a recording with a large number of exceptions? I realized that there might be some slowness with splitting this out if we're not careful about it.

Reply:
Looking good: https://app.replay.io/recording/exceptions--411587ca-3524-4653-8af3-10ac1aab14a1

Reply:
See here: https://app.replay.io/recording/exceptions--acf98816-1229-409a-8865-a0f7f8df624a
It's a lot slower than with the old Analysis API but I'd say our frontend handles that slowness rather well.